overlord/snapstate/storehelpers.go

```diff
@@ -230,6 +230,39 @@ func setActionValidationSetsAndRequiredRevision(action *store.SnapAction, valset
 	}
 }
 
+func downloadInfo(ctx context.Context, st *state.State, name string, revOpts *RevisionOptions, userID int, deviceCtx DeviceContext) (store.SnapActionResult, error) {
+	curSnaps, err := currentSnaps(st)
+	if err != nil {
+		return store.SnapActionResult{}, err
+	}
+
+	user, err := userFromUserID(st, userID)
+	if err != nil {
+		return store.SnapActionResult{}, err
+	}
+
+	opts, err := refreshOptions(st, nil)
+	if err != nil {
+		return store.SnapActionResult{}, err
+	}
+
+	action := &store.SnapAction{
+		Action:       "download",
+		InstanceName: name,
+	}
+
+	if revOpts != nil {
+		action.Revision = revOpts.Revision
+	}
```

Comment:
shouldn't this take into account channel and cohort as well? otherwise it should probably error if those are set and report they are unsupported atm